cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/config/oidc/ClientConfiguration.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.cloudgatewayservice.config.oidc;
+
+import jakarta.annotation.PostConstruct;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Data
+@Component
+@ConfigurationProperties(prefix = "spring.security.oauth2.client", ignoreInvalidFields = true)
+public class ClientConfiguration {
+
+    private static final String SYSTEM_ENV_PREFIX = "ZWE_configs_spring_security_oauth2_client_";
+    private static final Pattern REGISTRATION_ID_PATTERN = Pattern.compile(
+        "^" + SYSTEM_ENV_PREFIX + "([^_]+)_.*$"
+    );
+
+    private Map<String, Registration> registration = new HashMap<>();
```

Comment:
maybe can be changed to `Collections.unmodifiableMap` if the map won't be modified in other places?